Tests/GRPCTests/AsyncAwaitSupport/AsyncIntegrationTests.swift

```diff
@@ -0,0 +1,153 @@
+    super.tearDown()
+  }
+
+  func testUnary() {
+    XCTAsyncTest {
+      let get = self.echo.makeGetCall(.with { $0.text = "hello" })
+
+      let initialMetadata = try await get.initialMetadata
+      initialMetadata.assertFirst("200", forName: ":status")
+
+      let response = try await get.response
+      XCTAssertEqual(response.text, "Swift echo get: hello")
+
+      let trailingMetadata = try await get.trailingMetadata
+      trailingMetadata.assertFirst("0", forName: "grpc-status")
+
+      let status = await get.status
+      XCTAssertTrue(status.isOk)
+    }
+  }
+
+  func testClientStreaming() {
+    XCTAsyncTest {
+      let collect = self.echo.makeCollectCall()
+
+      try await collect.sendMessage(.with { $0.text = "boyle" })
+      try await collect.sendMessage(.with { $0.text = "jeffers" })
+      try await collect.sendMessage(.with { $0.text = "holt" })
+      try await collect.sendEnd()
+
+      let initialMetadata = try await collect.initialMetadata
+      initialMetadata.assertFirst("200", forName: ":status")
+
+      let response = try await collect.response
+      XCTAssertEqual(response.text, "Swift echo collect: boyle jeffers holt")
+
+      let trailingMetadata = try await collect.trailingMetadata
+      trailingMetadata.assertFirst("0", forName: "grpc-status")
+
+      let status = await collect.status
+      XCTAssertTrue(status.isOk)
+    }
+  }
+
+  func testServerStreaming() {
+    XCTAsyncTest {
+      let expand = self.echo.makeExpandCall(.with { $0.text = "boyle jeffers holt" })
+
+      let initialMetadata = try await expand.initialMetadata
+      initialMetadata.assertFirst("200", forName: ":status")
+
+      let respones = try await expand.responses.map { $0.text }.collect()
+      XCTAssertEqual(respones, [
+        "Swift echo expand (0): boyle",
+        "Swift echo expand (1): jeffers",
+        "Swift echo expand (2): holt",
+      ])
+
+      let trailingMetadata = try await expand.trailingMetadata
+      trailingMetadata.assertFirst("0", forName: "grpc-status")
+
+      let status = await expand.status
+      XCTAssertTrue(status.isOk)
+    }
+  }
+
+  func testBidirectionalStreaming() {
+    XCTAsyncTest {
+      let update = self.echo.makeUpdateCall()
+
+      var responseIterator = update.responses.map { $0.text }.makeAsyncIterator()
+
+      for (i, name) in ["boyle", "jeffers", "holt"].enumerated() {
+        try await update.sendMessage(.with { $0.text = name })
+        let response = try await responseIterator.next()
+        XCTAssertEqual(response, "Swift echo update (\(i)): \(name)")
+      }
+
+      try await update.sendEnd()
+
+      let initialMetadata = try await update.initialMetadata
+      initialMetadata.assertFirst("200", forName: ":status")
+
+      let trailingMetadata = try await update.trailingMetadata
+      trailingMetadata.assertFirst("0", forName: "grpc-status")
+
+      let status = await update.status
+      XCTAssertTrue(status.isOk)
```

Comment:
We have to `try` the trailingMetadata, but not the status? Is this intended? When do we get the status?

Reply:
It is intended but it's something I'm not sure of API-wise going forward. In the ELF-flavoured API the status promise will always be succeeded. We receive the status at the end of every RPC -- even it doesn't complete (we synthesise one locally if that's the case).